app/controllers/books_controller.rb

```diff
@@ -45,11 +45,18 @@ def by_category
     @category = ::Category.find_by(name: params[:name])
   end
 
-  def by_category
-    @category = ::Category.find_by(name: params[:name])
-  end
 
   private
+  def adult?
+    @user = current_user
+    (Time.now.year - @user.birghtday.year)>19
+  end
+  def for_grownups?
+    @book = Book.find(params[:id])
```

Comment:
Podobnie jak wyżej, jeśli nie używasz `@book` poza tą metodą, to możesz zrezygnować z linijki `55` i dać tylko:

```
def for_grownups?
  Book.find(params[:id]).category.for_grownups?
end
```

W innym przypadku - `@book` zapewni Ci before_action.